collectors/python.d.plugin/gearman/gearman.chart.py

```diff
@@ -0,0 +1,218 @@
+# Description: dovecot netdata python.d module
+# Author: Kyle Agronick (agronick)
+# SPDX-License-Identifier: GPL-3.0+
+
+# Gearman Netdata Plugin
+
+from bases.FrameworkServices.SocketService import SocketService
+from copy import deepcopy
+
+priority = 60500
+
+
+CHARTS = {
+    'total_workers': {
+        'options': ['total', 'Total Workers', 'Workers', 'Total Workers', 'gearman.total', 'line'],
+        'lines': [
+            ['total_pending', 'Pending', 'absolute'],
+            ['total_running', 'Running', 'absolute'],
+        ]
+    },
+}
+
+
+def job_chart_template(job_name):
+    return {
+        'options': [job_name, job_name, 'Workers', 'Jobs', 'gearman.job', 'stacked'],
+        'lines': [
+            ['{0}_pending'.format(job_name), 'Pending', 'absolute'],
+            ['{0}_idle'.format(job_name), 'Idle', 'absolute'],
+            ['{0}_running'.format(job_name), 'Running', 'absolute'],
+        ]
+    }
+
+def build_result_dict(job):
+    """
+    Get the status for each job
+    :return: dict
+    """
+
+    total, running, available = job['metrics']
+
+    idle = available - running
+    pending = total - running
+
+    return {
+        '{0}_pending'.format(job['job_name']): pending,
+        '{0}_idle'.format(job['job_name']): idle,
+        '{0}_running'.format(job['job_name']): running,
+    }
+
+def parse_worker_data(job):
+    job_name = job[0]
+    job_metrics = job[1:]
+
+    return {
+        'job_name': job_name,
+        'metrics': job_metrics,
+    }
+
+class Service(SocketService):
+    def __init__(self, configuration=None, name=None):
+        super(Service, self).__init__(configuration=configuration, name=name)
+        self.request = "status\n"
+        self._keep_alive = True
+
+        self.host = self.configuration.get('host', 'localhost')
+        self.port = self.configuration.get('port', 4730)
+
+        self.tls = self.configuration.get('tls', False)
+        self.cert = self.configuration.get('cert', None)
+        self.key = self.configuration.get('key', None)
+
+        self.active_jobs = set()
+        self.definitions = deepcopy(CHARTS)
+        self.order = ['total_workers']
+
+    def _get_data(self):
+        """
+        Format data received from socket
+        :return: dict
+        """
+
+        active_jobs = self.get_active_jobs()
+        found_jobs, job_data = self.process_jobs(active_jobs)
+        self.remove_stale_jobs(found_jobs)
+
+        if job_data:
+            return job_data
```

Comment:
i see it is always `job_data`, at least we always have

```
        output = {
            'total_pending': 0,
            'total_idle': 0,
            'total_running': 0,
        }
```

Reply:
You're right. Should have caught that one.

Reply:
use `return job_data or None`, it is more clear then implicit return

Reply:
btw i think that 
```
        output = {
            'total_pending': 0,
            'total_idle': 0,
            'total_running': 0,
        }
```

is ok, otherwise there will be no metrics if there is no jobs

Reply:
Don't you want no metrics with no jobs?

Reply:
No no, no job - zero total metrics, no metrics only in case of error